tests/trainer/test_trainer.py

```diff
@@ -39,6 +43,17 @@
 from tests.test_state import assert_state_equivalent
 
 
+class SleepyCallback(Callback):
```

Comment:
Should we add a docstring on what this does, given that it is public facing? Just one line should work.